src/stacks-hierarchy/saved-payment-utils.ts

```diff
@@ -4,42 +4,31 @@ import {storage} from '@atb/storage';
 import Bugsnag from '@bugsnag/react-native';
 import {useAuthState} from '@atb/auth';
 import {useListRecurringPaymentsQuery} from '@atb/ticketing/use-list-recurring-payments-query';
+import {RecurringPayment} from '@atb/ticketing';
 
-export function usePreviousPaymentMethod(): SavedPaymentOption | undefined {
-  const [paymentMethod, setPaymentMethod] = useState<
-    SavedPaymentOption | undefined
-  >(undefined);
+export function usePreviousPaymentMethods(): {
+  recurringPayments: RecurringPayment[] | undefined;
+  previousPaymentMethod: SavedPaymentOption | undefined;
+} {
   const {userId} = useAuthState();
-
   const {data: recurringPayments} = useListRecurringPaymentsQuery();
+  const [previousPaymentMethod, setPreviousPaymentMethod] =
+    useState<SavedPaymentOption>();
 
   useEffect(() => {
-    async function run(uid: string) {
-      const savedMethod = await getPreviousPaymentMethodByUser(uid);
-      if (
-        recurringPayments &&
-        savedMethod &&
-        savedMethod.savedType === 'recurring'
-      ) {
-        const method = recurringPayments.find(
-          (recurringPayment) =>
-            recurringPayment.id === savedMethod.recurringCard.id,
-        );
-        if (method) setPaymentMethod(savedMethod);
-      } else {
-        setPaymentMethod(savedMethod);
-      }
-    }
-
     if (!userId) {
-      setPaymentMethod(undefined);
-      return;
+      setPreviousPaymentMethod(undefined);
     } else {
-      run(userId);
+      getPreviousPaymentMethodByUser(userId).then((method) => {
+        setPreviousPaymentMethod(method);
```

Comment:
Right here just before setting it, might be a good place to ensure that this is still a valid option? I guess this is what @reidzeibel has already commented on.

Reply:
Edit: I see this has just been addresed in the new commit

Reply:
Yup, I missed a bit here. This will _also_ change in https://github.com/AtB-AS/mittatb-app/compare/rosvik/update-recurring-payments...rosvik/refactor-payment-types?expand=1 😄

Reply:
Btw just to mention it - the current validation ignores Vipps, which means that right now you never get Vipps preselected? Would be a nice feature to add.

Reply:
True! Good catch, me fix.